packages/wrangler/src/dev/miniflare.ts

```diff
@@ -355,8 +394,88 @@ export function buildMiniflareBindingOptions(config: MiniflareBindingsConfig): {
 				.join("\n"),
 	};
 
+	// Setup service bindings to external services
+	const serviceBindings: NonNullable<WorkerOptions["serviceBindings"]> = {
+		...config.serviceBindings,
+	};
+	for (const service of config.services ?? []) {
+		if (service.service === config.name) {
+			// If this is a service binding to the current worker, don't bother using
+			// the dev registry to look up the address, just bind to it directly.
+			serviceBindings[service.binding] = getName(config);
+			continue;
+		}
+
+		const target = config.workerDefinitions?.[service.service];
+		if (target?.host === undefined || target.port === undefined) {
+			// If the target isn't in the registry, always return an error response
+			// TODO(someday): could we do some sort of `Proxy`-prototype thing here
+			//  for a nice error message if a user tried to call an RPC method?
+			serviceBindings[service.binding] = () => {
+				const message = `[wrangler] Couldn't find \`wrangler dev\` session for service "${service.service}" to proxy to`;
```

Comment:
What makes `Response` here make more sense than blowing up with a `UserError`? Is one more graceful than the other, and/or is can we just not recover from a certain types of these failures?

Reply:
Thoughtful and justified. Nice!